src/Components/LanguageSelect/LanguageSelect.js

```diff
@@ -32,49 +32,98 @@ const useStyles = makeStyles(theme => ({
     },
     pad: {
         padding: theme.spacing(1)
+    },
+    autocomplete: {
+        color: theme.palette.primary.main,
+        "& .MuiOutlinedInput-notchedOutline": {
+            borderColor: theme.palette.primary.main
+        },
+        "&:hover .MuiOutlinedInput-notchedOutline": {
+            borderColor: theme.palette.primary.light
+        },
+        "&.Mui-focused .MuiOutlinedInput-notchedOutline": {
+            borderColor: theme.palette.primary.dark
+        }
+    },
+    label: {
+        color: theme.palette.primary.main
     }
 }));
 
-function LanguageSelect({ setLanguage }) {
+function LanguageSelect() {
     const classes = useStyles(Theme);
 
-    const [loading, setLoading] = useState(false);
+    const [loading, setLoading] = useState(true);
     const [languages, setLanguages] = useState([]);
 
+    const [selectedLanguage, setSelectedLanguage] = useState();
+    const { profile, updateSecondLanguage } = UserProfile();
+
+    function handleChange(e, v) {
+        updateSecondLanguage(v).then();
+    }
+
     async function getLanguages() {
         setLoading(true);
-        return await Promise.resolve(listLanguages())
+        return await Promise.resolve(ListLanguages())
             .then(l => setLanguages(l))
             .finally(() => setLoading(false));
     }
 
     useEffect(() => {
         getLanguages().finally();
     }, []);
-
+    useEffect(() => {
+        if (
+            profile &&
+            Boolean(Object.keys(profile).length) &&
+            Boolean(Object.keys(profile).includes("secondLanguage")) &&
+            Boolean(
+                Object.keys(profile["secondLanguage"]).includes("code") &&
+                    languages &&
+                    languages.some(
+                        l => l.code === profile["secondLanguage"].code
+                    )
+            )
+        ) {
+            setSelectedLanguage(
+                languages.filter(
+                    l => l.code === profile["secondLanguage"].code
+                )[0]
+            );
+        }
+    }, [profile, languages]);
     return (
         <>
             {loading ? (
                 <div className={clsx(classes.pad)}>
-                    <CircularProgress color="secondary" />
+                    <LoadingBar label="Languages" />
                 </div>
             ) : (
                 languages &&
-                Boolean(languages.length) && (
+                Boolean(languages.length) &&
+                Boolean(selectedLanguage) && (
```

Comment:
I believe for a new user, the value of `selectedLanguage` will be undefined, causing the HTML fragment to never show up. At the very least, removing this conditional clause allowed me to see the selector, pick a language, at which point I could re-add the conditional.